### PR TITLE
Assorted cleanups to schema (mostly around functions)

### PR DESCRIPTION
**Pull request by elprans**
See individual commits

### CODE REVIEW

File: edb/edgeql/compiler/stmt.py [hunk 1]
```diff
@@ -587,9 +587,8 @@ def init_stmt(
     ctx.stmt = irstmt
     if ctx.toplevel_stmt is None:
         parent_ctx.toplevel_stmt = ctx.toplevel_stmt = irstmt
-        ctx.path_scope = parent_ctx.path_scope
-    else:
-        ctx.path_scope = parent_ctx.path_scope.attach_fence()
+
+    ctx.path_scope = parent_ctx.path_scope.attach_fence()
```
Comment (1st1) [inline edb/edgeql/compiler/stmt.py:591]: > they would end up sharing the same scope node inappropriately.

Can we have a test for that?

File: edb/schema/functions.py [hunk 2]
```diff
@@ -403,53 +430,6 @@ class DeleteParameter(ParameterCommand, sd.DeleteObject[Parameter]):
     pass
 
 
-class PgParams(NamedTuple):
```
Comment (1st1) [inline edb/schema/functions.py:406]: Finally!

File: edb/edgeql/compiler/stmt.py (hunk 1 above)
Comment (elprans) [reply edb/edgeql/compiler/stmt.py:591]: Basically all function tests are the test for that with this rework, hence the change.

**Comment by tailhook** [comment]
Merged to publish a PR on top of this

**Pull request merged**